### PR TITLE
Decouple CI from build

### LINKED ISSUE #77: Handle secrets in `debian/rules`
Currently the CI handles secrets directly in its own custom steps, and the package will be unable to build without them.
A better solution would be for the files to always exist (but empty), so that the package can be built if the repo is forked.
This also means that workflow files can remain uniform.

### PR DESCRIPTION
**Pull request by m-roberts**
Closes #77

TODO: Add delimiting logic to allow support for environment variables that contain newlines
```
      - name: Build Debian package
        uses: pi-top/debian-package-build-action@master
        with:
          target_architecture: ${{ env.TARGET_ARCHITECTURE }}
          docker_image: ${{ env.DEB_BUILD_DOCKER_IMAGE }}:${{ env.DEB_BUILD_DOCKER_TAG }}
          build_directory: ./artifacts
          additional_env: |
            DATA=${{ secrets.DATA }}
            #NEWLINE#
            TLS_KEY=${{ secrets.TLS_KEY }}
          additional_env_delimiter: #NEWLINE#
```

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -7,7 +7,15 @@ include /usr/share/dpkg/pkg-info.mk
 
 override_dh_auto_build:
 	echo "__version__ = '$(DEB_VERSION)'" > src/lib/further_link/version.py
+	echo "$(DATA)" > extra/fl.dat
+	echo "$(TLS_KEY)" > extra/key.aes.pem
```
Comment (angusjfw) [inline debian/rules:11]:
```suggestion
	echo "$(SECRET_DATA)" > extra/fl.dat
	echo "$(SECRET_TLS_KEY)" > extra/key.aes.pem
```

File: .github/workflows/deb-build.yml [hunk 2]
```diff
@@ -55,11 +46,14 @@ jobs:
...
       - name: Build Debian package
-        uses: pi-top/debian-package-build-action@master
+        uses: pi-top/debian-package-build-action@additional_env_newline_parsing
         with:
           target_architecture: ${{ env.TARGET_ARCHITECTURE }}
           docker_image: ${{ env.DEB_BUILD_DOCKER_IMAGE }}:${{ env.DEB_BUILD_DOCKER_TAG }}
           build_directory: ./artifacts
+          additional_env: |
+            DATA=${{ secrets.DATA }}
+            TLS_KEY=${{ secrets.TLS_KEY }}
```
Comment (jcapona) [inline .github/workflows/deb-build.yml:56]:
```suggestion
            DATA="${{ secrets.DATA }}"
            TLS_KEY="${{ secrets.TLS_KEY }}"
```

File: debian/rules [hunk 3]
```diff
@@ -7,7 +7,15 @@ include /usr/share/dpkg/pkg-info.mk
 
 override_dh_auto_build:
 	echo "__version__ = '$(DEB_VERSION)'" > src/lib/further_link/version.py
+	[[ -f extra/fl.dat ]] || echo "$(DATA)" > extra/fl.dat
+	[[ -f extra/key.aes.pem ]] || echo -e "$(TLS_KEY)" > extra/key.aes.pem
```
Comment (jcapona) [inline debian/rules]:
```suggestion
	[[ -f extra/key.aes.pem ]] || echo -e $(TLS_KEY) > extra/key.aes.pem
```

File: debian/rules [hunk 4]
```diff
@@ -7,7 +7,15 @@ include /usr/share/dpkg/pkg-info.mk
 
 override_dh_auto_build:
 	echo "__version__ = '$(DEB_VERSION)'" > src/lib/further_link/version.py
+	[ -f extra/fl.dat ] || echo "$(DATA)" > extra/fl.dat
+	[ -f extra/key.aes.pem ] || echo -e "$(TLS_KEY)" > extra/key.aes.pem
```
Comment (jcapona) [inline debian/rules:11]:
```suggestion
	[ -f extra/key.aes.pem ] || echo -e $(TLS_KEY) > extra/key.aes.pem
```

File: src/util/ssl_context.py [hunk 5]
```diff
@@ -20,7 +20,7 @@ def ssl_context():
     cert = os.path.join(file_dir, '../../cert.pem')
     own_key = os.path.join(file_dir, '../../key.pem')
     encrypted_key = os.path.join(file_dir, '../../key.aes.pem')
-    data_file = os.path.join(file_dir, '../../data.txt')
+    data_file = os.path.join(file_dir, '../../fl.dat')
```
Comment (angusjfw) [inline src/util/ssl_context.py:23]: these need to be updated to look in extra/

**Review by angusjfw — APPROVED** [review]

**Review by jcapona — APPROVED** [review]

**Pull request merged**